src/atlas.nim

```diff
@@ -457,6 +458,57 @@ proc listOutdated(c: var AtlasContext) =
     listOutdated c, c.depsDir
   listOutdated c, c.workspace
 
+proc newProject(c: var AtlasContext; projectName: string) =
+  ## Tries to create a new project directory in the current dir
+  ## with a single bare `projectname.nim` file inside.
+  ## `projectName` is validated.
+
+  proc isValidProjectName(n: openArray[char]): bool =
+    ## Validates `n` as a project name:
+    ## Valid Nim identifier with addition of dashes (`-`) being allowed,
+    ## but replaced with underscores (`_`) for the `.nim` file name.
+    ## .. Note: Doesn't check if `n` is a valid file/directory name.
+    if not (n.len > 0 and n[0] in IdentStartChars):
+      return false
+    var prevDashOrUnderscore = false
+    for c in n:
+      if c in {'-', '_'}:
+        if prevDashOrUnderscore:
+          return false
+        else:
+          prevDashOrUnderscore = true
+      elif c notin (Letters + Digits):
+        return false
+      else:
+        prevDashOrUnderscore = false
+    true
+
+  let name = block:
```

Comment:
```suggestion
  let name = projectName.strip()
```

etc... No need for a block here.

Reply:
Thanks, good catch! I usually just start there to be safe with the compiler guarding me and don't always notice that all unhappy paths quit early when I'm done.